include/led.h

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ZEPHYR_LED_H
+#define _ZEPHYR_LED_H
+
+/**
+ * @file
+ * @brief Public LED driver APIs
+ */
+
+#include <zephyr/types.h>
+#include <device.h>
+
+/**
+ * @brief Driver specific representation of a LED device
+ */
+
+struct led_data {
```

Comment:
This isn't accessible through any of the core API routines. What's going on?

Reply:
I intended to use this struct inside the driver specific dev_data, which means not publicly available to the user. But anyway following your comments I'm going to modify this struct to be used by the API directly.

Reply:
I looked at this again. I do think the definition of this struct ought to be moved to the LP3943 driver, since other LED implementations may not share the same constraints.

Think about a GPIO-based driver where blink is implemented in a work queue. There would then be no limit on the maximum blink time, since you can always sleep as long as necessary between toggles.